homeassistant/components/foscam/config_flow.py

```diff
@@ -28,14 +33,23 @@
         vol.Required(CONF_USERNAME): str,
         vol.Required(CONF_PASSWORD): str,
         vol.Required(CONF_STREAM, default=STREAMS[0]): vol.In(STREAMS),
+        vol.Required(CONF_RTSP_PORT, default=DEFAULT_RTSP_PORT): int,
     }
 )
 
 
+def camera_unique_id(data):
+    """Generate an unique id for the camera."""
+    # MAC Address is only available with admin accounts so we craft our own unique id
+    return (
+        f"{data[CONF_USERNAME]}_{data[CONF_HOST]}:{data[CONF_PORT]}_{data[CONF_STREAM]}"
+    )
+
+
 class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow for foscam."""
 
-    VERSION = 1
+    VERSION = 3
```

Comment:
Why three and not two?

Reply:
This was because I added two fixes to the PR and treated both migrations separately, set it as 2 and merged the migration code for both versions.